app/components/profile/index.tsx

```diff
@@ -46,12 +46,34 @@ const mockWalletAddress = "0x822408EAC8C331002BE00070AFDD2A5A02065D3F";
 
 class ProfileContainer extends React.PureComponent<IProfileContainerProps, {}> {
   public componentDidMount() {
-    const { dispatch, currentUserState } = this.props;
-    const { profileImage, institution, major } = currentUserState;
-
-    dispatch(Actions.syncSettingInputWithCurrentUser(profileImage, institution, major));
+    const { dispatch, currentUserState, match } = this.props;
+    const { id, profileImage, institution, major } = currentUserState;
+    const userId = match.params.userId;
+    // TODO : asynchronous
+    if (id === parseInt(userId, 10)) {
+      dispatch(Actions.syncSettingInputWithCurrentUser(profileImage, institution, major));
+      dispatch(Actions.syncCurrentUserWithProfileUser(currentUserState));
+    } else {
+      dispatch(Actions.getUserProfile(userId));
+    }
   }
 
+  public componentWillReceiveProps(nextProps: IProfileContainerProps) {
+    const beforeParamUserId = this.props.match.params.userId;
+    const nextParamUserId = nextProps.match.params.userId;
+    const currentUserStateId = this.props.currentUserState.id;
+    const { dispatch } = this.props;
+    const { profileImage, institution, major } = this.props.currentUserState;
+
+    if (beforeParamUserId !== nextParamUserId) {
+      if (currentUserStateId === parseInt(nextParamUserId, 10)) {
+        dispatch(Actions.syncSettingInputWithCurrentUser(profileImage, institution, major));
```

Comment:
The fetching method logic is similar with upper one. use common private instance method.